### PR TITLE
[Quantization] Add "quantization_tag" as metadata to fx proxy

### PR DESCRIPTION
**Pull request by kimishpatel**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #108764

Summary:
In order to make sure that quantization_tag is preserved through second
stage export, this PR adds it as a special metadata that should be
preserved.

Since quantization in export path will work on top of pre dispatch
graph, subsequent post dispatch op decomposition, will decompose ops
that quant workflow tagged. In order to make sure that the patterns
identified by quantizer, remains identifiable, even after decompositions
are applied, we must preserve "quantization_tag".

This enables backend delegates, that quantized a model for specific
backend, to be able to identify "quantized" patterns.

Test Plan:
metadata porting tests

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D49056259](https://our.internmc.facebook.com/intern/diff/D49056259)

### CODE REVIEW

**Comment by kimishpatel** [comment]
@kimishpatel has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D49056259).

**Comment by tugsbayasgalan** [comment]
I have a potentially dumb question: "How is this different from using source_fn to do pattern matching?"

**Comment by kimishpatel** [comment]
> I have a potentially dumb question: "How is this different from using source_fn to do pattern matching?"

source_fn based pattern matching doesnt allow for arbitrary pattern. It is related to nn.Module or nn.function (functinal actually doesnt work). So if you have some fusion patterns it doesnt work. Plus this specific taag enables quant workflow to prot metadata for nodes that quantization adds

**Review by tugsbayasgalan — APPROVED** [review]

File: test/quantization/pt2e/test_metadata_porting.py [hunk 1]
```diff
@@ -146,13 +177,21 @@ def validate(self, model: torch.fx.GraphModule) -> None:
...
-        self._test_metadata_porting(
+        m = self._test_metadata_porting(
             TestHelperModules.Conv2dWithObsSharingOps(),
             example_inputs,
             BackendAQuantizer(),
             node_tags,
         )
 
+        from_node_to_tags = {
+            torch.ops.aten.adaptive_avg_pool2d.default: "BackendA_adaptive_avg_pool2d_0",
```
Comment (jerryzh168) [inline test/quantization/pt2e/test_metadata_porting.py:188]: should we test metadata on get_attr nodes?

**Comment by kimishpatel** [comment]
@kimishpatel has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D49056259).

**Review by jerryzh168 — APPROVED** [review]

**Comment by kimishpatel** [comment]
@kimishpatel has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D49056259).

**Pull request closed without merging**